src/pg/sql/11_cdb_union_adjacent.sql

```diff
@@ -0,0 +1,43 @@
+CREATE OR REPLACE FUNCTION _cdb_final_union_adjacent( joined_geoms geometry[] )
+RETURNS geometry[] AS $$
+BEGIN
+    RETURN joined_geoms;
+END
+$$ LANGUAGE plpgsql;
+
+
+CREATE OR REPLACE FUNCTION _cdb_state_update_union_adjacent(clusters geometry[], new_geom  geometry)
+RETURNS geometry[] AS $$
+DECLARE
+  joins  geometry[] :='{}';
+  unjoined geometry[] :='{}';
+  i integer;
+  combined geometry;
+BEGIN
+  joins := (select array_agg(g)
```

Comment:
`joins` and `unjoined` could be generated in one `select`.

One way to do so would be to `array_agg(CASE WHEN ST_TOUCHES(g, new_geom) THEN g ELSE NULL END)` and the same for the opposite case as a second column.